R/create_protocol.R

```diff
@@ -148,31 +227,96 @@ create_sfp <- function(
                  procedure = protocol_code,
                  theme = theme,
                  book_filename = book_filename,
-                 output_dir = output_dir
+                 output_dir = output_dir_rel
     )
     writeLines(map_chr(original_file_content,
                        whisker.render,
                        data),
                filename)
     close(original_file)
   }
+
+  # docx
+  if (!is.null(from_docx)) {
+    convert_docx_to_rmd(
+      from = from_docx,
+      to = book_filename,
+      dir = path_to_protocol,
+      wrap = 80,
+      overwrite = FALSE,
+      verbose = FALSE)
+    # convert emf to png
+    emf_images <- list.files(path = file.path(path_to_protocol, "media"),
+                             pattern = ".emf",
+                             full.names = TRUE)
+    for (img in emf_images) {
+      img_emf <- image_read(path = img)
+      image_write(image = img_emf,
+                  format = "png",
+                  path = str_replace(img, ".emf", ".png"))
+      file.remove(img)
+    }
+    # move relevant sections
+    contents <- readLines(con = file.path(path_to_protocol,
+                                          book_filename))
+    contents <- str_replace_all(contents, ".emf", ".png")
+    is_title <- str_detect(string = contents, pattern = "^(#{1}\\s{1})")
+    title_numbers <- formatC(x = cumsum(is_title),
+                             width = 2, format = "d", flag = "0")
+    filenames <- str_remove(string = tolower(contents[is_title]),
+                            pattern = "^(#{1}\\s{1})")
+    filenames <- str_remove(string = filenames,
+                            pattern = "\\s$")
+    filenames <- str_replace_all(filenames, pattern = "\\s", replacement = "_")
+    filenames <- paste0(unique(title_numbers), "_", filenames, ".Rmd")
+    # delete the empty template chapters
+    file.remove(
+      list.files(
+        path = path_to_protocol,
+        pattern = "^\\d{2}_",
+        full.names = TRUE
+      )
+    )
+    # create new chapters
+    file.create(file.path(path_to_protocol, filenames))
+    # and add chapter contents from docx
+    for (chapter in unique(cumsum(is_title))) {
+      chapter_file <- file.path(path_to_protocol, filenames[chapter])
+      chapter_contents <- contents[chapter == cumsum(is_title)]
+      writeLines(text = chapter_contents,
+                 con = chapter_file)
+    }
+  }
+  # delete the complete Rmd (output of convert_docx_rmd)
+  file.remove(file.path(path_to_protocol, book_filename))
+
   # render html
-  old_wd <- getwd()
-  setwd(path_to_protocol)
-  render_book(input = "index.Rmd")
-  setwd(old_wd)
+  if (render) {
+    old_wd <- getwd()
+    setwd(path_to_protocol)
+    render_book(input = "index.Rmd")
+    setwd(old_wd)
+  }
```

Comment:
@hansvancalster I suggest to put this part in a separate function, e.g. `render_protocol()`, for the user to be able to renew rendered protocols after manually updating existing Rmd files. To that aim, the first part of the `create_sfp()` function, needed to create `path_to_protocol`, is best wrapped in a separate (internal) function so that both `create_sfp()` and `render_protocol()` can make use of it.

Reply:
Done in #21